City-of-Helsinki/hel-fi-drupal-grants
# feat: AU-676: Permission changes

# [AU-676](https://helsinkisolutionoffice.atlassian.net/browse/AU-676),  [AU-1526](https://helsinkisolutionoffice.atlassian.net/browse/AU-1526)

## What was done

This pull request implements permission changes to the following roles: `Content producer`, `Grants producer` and `Grants producer INDUSTRY`. A new role called `Content producer INDUSTRY` has also been added. The roles and their new permissions are listed below.

### Role: _Content producer_

#### Nodes
- Can create, edit, translate and unpublish **any** `Service page`. 
- Can create, edit, translate and unpublish **any** `Landing page`.
- Can create, edit, translate and unpublish **any** `Standard page`.
- Can delete **own** `Service page` nodes and translations.
- Can delete **own** `Landing page` nodes and translations.
- Can delete **own** `Standard page` nodes and translations.

#### Taxonomy terms
- Can create, edit, translate and unpublish **any** `Avustuslaji` terms.
- Can create, edit, translate and unpublish **any** `Target group` terms.

#### Media
- Can create, edit, translate and unpublish **any** `Image` media.
- Can create, edit, translate and unpublish **any** `Chart` media.
- Can create, edit, translate and unpublish **any** `File` media.
- Can create, edit, translate and unpublish **any** `Remote video` media.
- Can delete **own** `Image` media and translations.
- Can delete **own** `Chart` media and translations.
- Can delete **own** `File` media and translations.
- Can delete **own** `Remote video` media and translations.

#### Menu
- Can add links to the main menu.


### Role: _Content producer INDUSTRY_

#### Nodes
- Can create, edit, translate and unpublish **any** `Service page` with a **matching industry**. 
- Can delete **own** `Service page` nodes and translations.

#### Taxonomy terms
- Can create, edit, translate and unpublish **any** `Avustuslaji` terms.
- Can create, edit, translate and unpublish **any** `Target group` terms.

#### Media
- Can create, edit, translate, unpublish and delete **own** `Image` media.
- Can create, edit, translate, unpublish and delete **own** `Chart` media.
- Can create, edit, translate, unpublish and delete **own** `File` media.
- Can create, edit, translate, unpublish and delete **own** `Remote video` media.

#### Menu
- Can add links to the main menu.

### Role: _Grants producer_
- Can edit the third-party settings on **any** `Webform`.

### Role: _Grants producer INDUSTRY_
- Can edit the third-party settings on a `Webform` with a **matching industry**.

### Other notable changes
- The module `admin_toolbar_links_access_filter` has been **enabled**. This removes all the unnecessary/dead links from the admin toolbar.
- The module `override_node_options` has been **downloaded** and **enabled**. The module allows us to remove the "Administer node" permission from all roles, and give more fine-tuned permission to roles instead. 
- The module `view_unpublished` has been **downloaded** and **enabled**. This allows content editors to view unpublished nodes.
- The module menu_admin_per_menu has been **downloaded** and **enabled**. This allows content editors to edit only the main menu and nothing else.

## How to install
Make sure your instance is up and running on the correct branch.
  * `git checkout feature/AU-676-permission-adjustment`
  * `make fresh`
  * `make drush-cr`

## How to test

The roles and their permissions can me tested in the dev environment. Instructions can be found here: https://helsinkisolutionoffice.atlassian.net/browse/AU-676.

- Review the ticket. Are the new roles and permissions in line with the ticket?
- Review the permissions at `/admin/people/permissions`. Do they look alright?
- Create a new test user and test all the roles and the permissions. Is everything working as listed above?
- Should anything be changed regarding the permissions?
- Review the code.

[AU-676]: https://helsinkisolutionoffice.atlassian.net/browse/AU-676?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[AU-1526]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1526?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
